### DISCUSSION TITLE
Get Results from a manually defined model

### DISCUSSION THREAD

**Discussion by PascalGitz** [Dlubal-Software/RFEM_Python_Client#288] [Q&A] [answered]
Hello

I am trying to get the member results from my model in my python script in an automated way. I would like to plot the function of the bending moments in matplotlib. So I need an array of the member results.

![image](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/115550508/46be1a71-1691-46bf-a528-c1c1387fdd4d)

## First approach
I've tried to connect to the model this way:

```
from RFEM.Results.resultTables import GetMaxValue, GetMinValue, ResultTables
from RFEM.initModel import Model, Calculate_all
# RFEM 6
Model(False, "TVS_Versuch_2_A3_Jaeger.rf6")
Model.clientModel.service.begin_modification()
Calculate_all()
results =  ResultTables.MembersInternalForces(
                                              loading_no=1,
                                              object_no=1)
`Model.clientModel.service.finish_modification()
```

which results in this error:

```
File [c:\Users\Pascal](file:///C:/Users/Pascal) Gitz\miniconda3\envs\calcs\lib\site-packages\RFEM\Results\resultTables.py:22, in GetResultTableParameters(results)
     20 params['base'] = list(set(params['base'] + i.__keylist__))
     21 if 'row' in i.__keylist__:
---> 22     params['row'] = list(set(params['row'] + i.row.__keylist__))
     23 else:
     24     params['errors'] = "Result table doesn't have attribute 'row'."

AttributeError: 'Text' object has no attribute '__keylist__'
```

## Second approach

Then i tried to query the model:

```
from suds.client import Client
from RFEM.enums import CaseObjectType

client = Client('http://localhost:8081/wsdl')
new = client.service.get_active_model()+'wsdl'
model = Client(new)

print(model.service.get_nodal_support(1))
print(model.service.get_results_for_members_internal_forces(loading_type='?????', loading_no=1, member_no = 1))
```
It works fine for getting nodal infos and other parameters. But I cannot get the results for member_internal_forces because i cannot figure out what to put in loading_type.

What approach is suited for my workflow? And how to do it :)?

**Comment by heetrojivadiya** [comment, marked as answer]
Hello @PascalGitz ,

I think that the problem occurs due to older version of RFEM package. We have fixed this bug in latest version of RFEM==v1.18.0. Please update RFEM package by using pip or you can clone main branch in your repository. This should fix your issue. 

For your second approach, loading type is enumeration name for CaseObjectType(for which you want to get member result). You can use ```E_OBJECT_TYPE_DESIGN_SITUATION, E_OBJECT_TYPE_LOAD_CASE, E_OBJECT_TYPE_LOAD_COMBINATION or E_OBJECT_TYPE_RESULT_COMBINATION``` in loading_type as a string input.

from my personal experience, I can suggest that first approach is user friendly while it is easy to change and read for different type of results. for second approach, try to store result table in one variable and print it rather than printing directly the result table. By doing this, it will work as first approach.

Let us know if you still face the same issue in first approach after updating RFEM package. 

Thanks and best regards,
Heet Rojivadiya
